onnx/defs/tensor/old.cc

```diff
@@ -3192,21 +3192,40 @@ ONNX_OPERATOR_SET_SCHEMA(
               starts.size() != ends.size()) {
             fail_shape_inference(
                 "Incorrect or missing attribute value for starts and ends");
-            ;
           }
+
+          auto is_negative = [](int64_t index) {
+            return index < 0;
+          };
+          bool has_negative_axis = false;
+          if (!std::none_of(starts.begin(), starts.end(), is_negative) ||
+              !std::none_of(ends.begin(), ends.end(), is_negative)) {
+            has_negative_axis = true;
+          }
+
           std::vector<int64_t> axes;
           if (!getRepeatedAttribute(ctx, "axes", axes)) {
             for (int i = 0; (size_t)i < starts.size(); ++i) {
               axes.push_back(i);
             }
           } else if (axes.size() != starts.size()) {
             fail_shape_inference("Attribute axes has incorrect length");
-            ;
+          } else if (!std::none_of(axes.begin(), axes.end(), is_negative)) {
```

Comment:
adding this check as else if does not seem right... both the if condition above and this else if condition regarding negative axes check can be true at once right?

Reply:
yes. I think we should replace `!std::none_of` by `std::any_of`

Reply:
sorry, my comment is a different from one Ashwini's. We may need to change the nested-else, but the condition also needs to be changed, I think

Reply:
It seems to work ok (since we fail in the previous case).

Reply:
aah right. I think only changing !none_of to any_of (for better readability) should be enough